application/client/src/app/ui/views/toolbar/chart/chart/chart.ts

```diff
@@ -0,0 +1,411 @@
+import { EChartName, ILabel, IPosition, EScaleType } from '../common/types';
+import {
+    BubbleDataPoint,
+    ChartDataset,
+    ChartTypeRegistry,
+    ScatterDataPoint,
+    Chart as CanvasChart,
+    ScaleOptionsByType,
+    CartesianScaleTypeRegistry,
+} from 'chart.js';
+import { ChangesDetector } from '@ui/env/extentions/changes';
+import { IlcInterface } from '@service/ilc';
+import { Session } from '@service/session';
+import { ChartRequest } from '@service/session/dependencies/search/charts/request';
+import { IRange } from '@platform/types/range';
+import { Service } from '../service';
+
+export class Chart {
+    private readonly _session: Session;
+    private readonly _parent: IlcInterface & ChangesDetector;
+    private readonly _service: Service;
+    private readonly _chart: CanvasChart;
+    private readonly _labelState: ILabel;
+    private _zoomedRange!: IRange;
+    private _canvasWidth: number = 0;
+    private _defaultPosition: IPosition = {
+        full: this._canvasWidth,
+        left: 0,
+        width: this._canvasWidth,
+    };
+
+    constructor(
+        session: Session,
+        parent: IlcInterface & ChangesDetector,
+        service: Service,
+        canvasWidth: number,
+        labelState: ILabel,
+    ) {
+        this._session = session;
+        this._parent = parent;
+        this._service = service;
+        this.canvasWidth = canvasWidth;
+        this._labelState = labelState;
+        this._chart = this._createChart();
+        this._initSubscriptions();
+        this._initDatasets();
+        this._initData();
+        this._restoreZoomedRange();
+    }
+
+    public set canvasWidth(canvasWidth: number) {
+        this._canvasWidth = canvasWidth;
+        this._defaultPosition = {
+            full: this._canvasWidth,
+            left: 0,
+            width: this._canvasWidth,
+        };
+    }
+
+    public get reverseScaleType(): EScaleType {
+        return (this._chart.options as any).scales['y'].type === EScaleType.linear
+            ? EScaleType.logarithmic
+            : EScaleType.linear;
+    }
+
+    public destroy() {
+        this._chart.destroy();
+    }
+
+    public switchScaleType() {
+        (this._chart.options as any).scales['y'].type = this.reverseScaleType;
+        this._chart.update();
+    }
+
+    public zoom(zoomedRange: IRange) {
+        this._zoomedRange = zoomedRange;
+        [
+            (this._chart.config.options as any).scales['x'].min,
+            (this._chart.config.options as any).scales['x'].max,
```

Comment:
is it possible to get rid of `any`? chartJS should have some type description, no?